### PR TITLE
Change form validation to support Bootstrap 4.0.0.beta (replaces #345)

### PR DESCRIPTION
**Pull request by CyberDeck**
This should contain all relevant changes to reassemble the old PR #345.

Bugfixes:
  - `form-control-danger` is replaced with `is-invalid` for bootstrap 4.0.0.beta3
  - `form-control-feedback` is replaced with `invalid-feedback` for bootstrap 4.0.0.beta3
  - help texts are rendered with `<small>` tag instead of `<span>` tag, i.e. like in bootstrap 4.0.0.beta3 

Features:
  - `is-valid` will be applied to validated form field (i.e. not on unchanged or new records)
  - new `custom: true` option for radio buttons and check boxes according to bootstrap 4.0.0.beta3

### CODE REVIEW

**Comment by CyberDeck** [comment]
Now this PR addresses the action
`Remove/replace has_danger, form-control-feedback, and form-control-label` from #361.

File: lib/bootstrap_form/form_builder.rb [hunk 1]
```diff
@@ -114,12 +114,24 @@ def time_zone_select_with_bootstrap(method, priority_zones = nil, options = {},
 
     def check_box_with_bootstrap(name, options = {}, checked_value = "1", unchecked_value = "0", &block)
       options = options.symbolize_keys!
-      check_box_options = options.except(:label, :label_class, :help, :inline)
-      check_box_options[:class] = ["form-check-input", check_box_options[:class]].compact.join(' ')
+      check_box_options = options.except(:label, :label_class, :help, :inline, :custom)
+      if options[:custom]
+        validation = nil
+        validation = "is-invalid" if has_error?(name)
+        validation = "is-valid" if is_valid?(name)
```
[the remark this answers is not in the document]
Comment (donv) [reply lib/bootstrap_form/form_builder.rb]: It seems like the right thing to, just as it was before, just with the new classes.

[the remark this answers is not in the document]
Comment (donv) [reply lib/bootstrap_form/form_builder.rb]: Sorry, didn't read the context of this comment.  Now I am not so sure.  I think marking the valid fields is OK, but only marking the invalid fields is better.  Actually, for red/green color blind users, only marking the invalid fields helps them stand out.

[the remark this answers is not in the document]
Comment (donv) [reply lib/bootstrap_form/form_builder.rb]: A quick test in a real app showed me that marking valid fields with green looks OK, but does not improve the UX, really.

[the remark this answers is not in the document]
Comment (mattbrictson) [reply lib/bootstrap_form/form_builder.rb]: I think we should leave out `is-valid` for this PR. I don't see myself using it for server-side validation.

[the remark this answers is not in the document]
Comment (CyberDeck) [reply lib/bootstrap_form/form_builder.rb]: I just removed the 'is-valid' class within the PR.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,14 @@
 
 Bugfixes:
   - Your contribution here!
+  - form-control-danger is replaced with is-invalid for bootstrap 4.0.0.beta3
+  - form-control-feedback is replaced with invalid-feedback for bootstrap 4.0.0.beta3
+  - help texts are rendered with <small> tag instead of <span> tag, i.e. like in bootstrap 4.0.0.beta3 
 
 Features:
   - Your contribution here!
+  - is-valid will be applied to validated form field (i.e. not on unchanged or new records)
```
Comment (donv) [inline CHANGELOG.md]: This is not true anymore, right?

Comment (CyberDeck) [reply CHANGELOG.md]: Yes. I will remove it.

File: lib/bootstrap_form/form_builder.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (desheikh) [reply lib/bootstrap_form/form_builder.rb]: Yup, definitely feels like the right behaviour. Thanks for the great work @CyberDeck!

**Comment by mattbrictson** [comment]
@donv @desheikh is this ready to merge?

**Comment by mattbrictson** [comment]
Thanks for the PR and congratulations on your first contribution to the project! 🎉

**Pull request merged**